CMakeLists.txt

```diff
@@ -2,51 +2,111 @@ project(libMCNP2CAD)
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
-    if(NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
+# Default to a release build
+if (NOT CMAKE_BUILD_TYPE)
+  message(STATUS "CMAKE_BUILD_TYPE not specified, defaulting to Release")
+  set(CMAKE_BUILD_TYPE Release)
+endif ()
+  
+# Use C++11
+set(CMAKE_CXX_STANDARD 11)
+
+# iGeom assumed to have cone functionality
+add_definitions(-DHAVE_IGEOM_CONE=ON)
+
+# Adjust compiler setting for Linux using gcc version 5.0 and higher
+if (CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
+    if (NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
       set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
-    endif()
-  endif()
-endif()
-
-if(${BUILD_CLI})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/cli)
-endif()
-
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
-
-# These directories need to be defined by the user.
-include_directories(${IGEOM_INCLUDE_DIR})
-find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_LIB_DIR})
-
-SET(LIB_SRC
-    mcnp2cad.cpp
-    mcnp2cad.hpp
-    MCNPInput.cpp
-    MCNPInput.hpp
-    geometry.cpp
-    geometry.hpp
-    volumes.cpp
-    volumes.hpp
-    GQ_Characterize.cpp
-    GQ_Characterize.hpp
-    test_GQ.cpp
-)
-
-add_library(mcnp2cad SHARED ${LIB_SRC})
-target_link_libraries(mcnp2cad ${IGEOM_LIB})
-
-add_executable(test_GQ test_GQ.cpp GQ_Characterize.cpp)
-target_link_libraries(test_GQ)
-if(IGEOM_LIB_DIR)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${IGEOM_LIB_DIR} BUILD_WITH_INSTALL_RPATH TRUE)
-elseif(CMAKE_PREFIX_PATH)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${CMAKE_PREFIX_PATH} BUILD_WITH_INSTALL_RPATH TRUE)
-endif()
-
-enable_testing()
-add_test(NAME test_gq_characterization COMMAND test_GQ)
+    endif ()
+  endif ()
+endif ()
+
+# Find iGeom. If mcnp2cad is being built as a standalone library, then
+# -DIGEOM_DIR must be specified. If it is not being built as a standalone
+# library, then it is assumed that the location of the iGeom library is known
+# and that the iGeom headers have already been included.
+if (STANDALONE_MCNP2CAD)
+  if (NOT IGEOM_DIR)
+    message(FATAL_ERROR "-DIGEOM_DIR must be specified when building mcnp2cad"
+                        " as a standalone library")
+  else ()
+    message(STATUS "Building standalone mcnp2cad")
+  endif ()
+
+  # Find location of iGeom headers
+  find_path(IGEOM_INCLUDE_DIR NAMES iGeom.h HINTS ${IGEOM_DIR}/include)
+  include_directories(${IGEOM_INCLUDE_DIR})
+  message(STATUS "IGEOM_INCLUDE_DIR: ${IGEOM_INCLUDE_DIR}")
+
+  # Find location of iGeom library and create an imported target for it
+  find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_DIR}/lib)
+  get_filename_component(IGEOM_LIB_DIR ${IGEOM_LIB} DIRECTORY)
+  link_directories(${IGEOM_LIB_DIR})
+  add_library(iGeom SHARED IMPORTED)
+  set_target_properties(iGeom PROPERTIES IMPORTED_LOCATION "${IGEOM_LIB}")
```

Comment:
Let's also set the `INCLUDE_DIRECTORIES` property of this target. This will remove the need for the `include_directories` call a few lines above.

Reply:
```suggestion
  set_target_properties(iGeom PROPERTIES 
                              IMPORTED_LOCATION "${IGEOM_LIB}"
                              INCLUDE_DIRECTORIES ${IGEOM_INCLUDE_DIR})
```

Reply:
This didn't work (it failed to find iGeom.h), but a slightly different version did:

```
target_include_directories(iGeom INTERFACE "${IGEOM_INCLUDE_DIR}")
...
target_link_directories(iGeom INTERFACE "${IGEOM_LIB_DIR}")
```

Reply:
Sorry, that property should've been `INTERFACE_INCLUDE_DIRECTORIES`. 

The `target_include_directories` and `target_link_directories` commands will only work for imported targets if the CMake version is > 3.12. Seeing as we require >= 2.8 for this project, I think we have to stick with the `set_target_properties` approach.

Reply:
I like the setting the `INTERFACE_LINK_DIRECTORIES`  of the `iGeom` btw. We should keep that as part of the `set_target_properties` call.

Reply:
Ok I changed this to use `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_DIRECTORIES` and verified that it works with cmake 2.8